components/camel-snmp/src/main/java/org/apache/camel/component/snmp/SnmpEndpoint.java

```diff
@@ -265,8 +265,8 @@ public void setProtocol(String protocol) {
     }
 
     @Override
-    protected void doStart() throws Exception {
-        super.doStart();
+    protected void doInit() throws Exception {
+        super.doInit();
 
         URI uri = URI.create(getEndpointUri());
```

Comment:
Oh there may be a problem when creating an URI as its a network resource. And its either URL or URI in Java that does some networking thingy when being created. @gnodet @ppalaga @lburgazzoli @oscerd @jamesnetherton ?

Reply:
No, there is no network access when creating a `URI` afaik.

Reply:
I don't think so

Reply:
Ah thanks yeah I found the answer too in the javadocs, URI mentions further below that URL on contrast does IO networking: https://docs.oracle.com/javase/7/docs/api/java/net/URI.html